packages/react-components/react-progress/src/components/Progress/useProgressStyles.ts

```diff
@@ -1,33 +1,386 @@
+        )`,
+        animationName: IndeterminateProgressRTL,
+        animationDuration: '3s',
+        animationIterationCount: 'infinite',
+
+        '@media screen and (forced-colors: active)': {
+          backgroundColor: 'Highlight',
+        },
+      },
+    },
+  },
+  indeterminateRTLLarge: {
+    width: '100%',
+    ['& > div.fui-Progress__Container']: {
+      position: 'relative',
+      ...shorthands.overflow('hidden'),
+      height: barThicknessValues.large,
+      ...shorthands.padding('8px'),
+
+      ['& > div.fui-Progress__Track']: {
+        position: 'absolute',
+        width: '100%',
+        height: barThicknessValues.large,
+        backgroundColor: tokens.colorNeutralBackground6,
+
+        '@media screen and (forced-colors: active)': {
+          ...shorthands.borderBottom('1px', 'solid', 'CanvasText'),
+        },
+      },
+
+      ['& > div.fui-Progress__Bar']: {
+        height: barThicknessValues.large,
+        position: 'absolute',
+        width: 0,
+        transitionProperty: 'width .3s ease',
+        minWidth: '33%',
+        backgroundImage: `linear-gradient(
+          to left,
+          ${tokens.colorNeutralBackground6} 0%,
+          ${tokens.colorCompoundBrandBackground} 50%,
+          ${tokens.colorNeutralBackground6} 100%
+        )`,
+        animationName: IndeterminateProgressRTL,
+        animationDuration: '3s',
+        animationIterationCount: 'infinite',
+
+        '@media screen and (forced-colors: active)': {
+          backgroundColor: 'Highlight',
+        },
+      },
+    },
+  },
 });
 
 /**
  * Apply styling to the Progress slots based on the state
  */
 export const useProgressStyles_unstable = (state: ProgressState): ProgressState => {
-  const styles = useStyles();
-  state.root.className = mergeClasses(progressClassNames.root, styles.root, state.root.className);
+  const { appearance, determinate = false, barThickness = 'default' } = state;
+  const rootStyles = useRootStyles();
+  const indicatorStyles = useIndicatorStyles();
+  const labelStyles = useLabelStyles();
+  const descriptionStyles = useDescriptionStyles();
+  const { dir } = useFluent();
+
+  state.root.className = mergeClasses(progressClassNames.root, rootStyles.root, state.root.className);
+
+  // Determinate indicator when determinate is defined
+  if (state.indicator) {
+    state.indicator.className = mergeClasses(
+      progressClassNames.indicator,
+      appearance === 'primary' &&
+        !determinate &&
+        dir === 'ltr' &&
+        barThickness === 'default' &&
+        indicatorStyles.indeterminateLTRDefault,
+      appearance === 'primary' &&
+        !determinate &&
+        dir === 'ltr' &&
+        barThickness === 'large' &&
+        indicatorStyles.indeterminateLTRLarge,
+      appearance === 'primary' &&
+        !determinate &&
+        dir === 'rtl' &&
+        barThickness === 'default' &&
+        indicatorStyles.indeterminateRTLDefault,
```

Comment:
Sorry, but what this does? Griffel automatically flips CSS values until they are CSS variables.

Even if some of them are not flippable, I would expect to see them as overrides for LTR rules rather than fully duplicated set.

Reply:
For RTL languages, the animation direction would be flipped as well, so this applies styling that flips the animation. I am unsure whether Griffel would do that automatically?

Reply:
> For RTL languages, the animation direction would be flipped as well, so this applies styling that flips the animation. I am unsure whether Griffel would do that automatically?

Correct, not all things are handled by Griffel, but:

```ts
makeStyles({
  indeterminateRTLDefault: {
    width: "100%", // no need to flip
    ["& > div.fui-Progress__Container"]: {
      position: "relative", // no need to flip
      ...shorthands.overflow("hidden"), // no need to flip
      height: barThicknessValues.default, // no need to flip
      ...shorthands.padding("8px"), // will be flipped

      ["& > div.fui-Progress__Track"]: {
        position: "absolute", // no need to flip
        width: "100%", // no need to flip
        height: barThicknessValues.default, // no need to flip
        backgroundColor: tokens.colorNeutralBackground6, // no need to flip

        "@media screen and (forced-colors: active)": {
          ...shorthands.borderBottom("1px", "solid", "CanvasText"), // no need to flip
        },
      },

      ["& > div.fui-Progress__Bar"]: {
        height: barThicknessValues.default, // no need to flip
        position: "absolute", // no need to flip
        width: 0, // no need to flip
        transitionProperty: "width .3s ease", // no need to flip
        minWidth: "33%", // no need to flip
        backgroundImage: `linear-gradient(
          to left,
          ${tokens.colorNeutralBackground6} 0%,
          ${tokens.colorCompoundBrandBackground} 50%,
          ${tokens.colorNeutralBackground6} 100%
        )`, // will be flipped 🎉
      },
    },
  },
});
```

```ts
makeStyles({
  root: {
    backgroundImage: `linear-gradient(to left, var(--foo) 0%, var(--bar) 50%, var(--baz) 100%)`,
  },
});
```

⬇️⬇️⬇️

```css
.f1ywtyre {
  background-image: linear-gradient(to left, var(--foo) 0%, var(--bar) 50%, var(--baz) 100%);
}

.fuw8sr5 {
  background-image: linear-gradient(to right, var(--foo) 0%, var(--bar) 50%, var(--baz) 100%);
}
```

Reply:
I suggest to check rules in [playground](https://griffel.js.org/try-it-out/) to see if they will flip properly 😉

Reply:
I played around with it, and it looks like it will animate properly if flipped, so I've changed the styling for RTL to just add `animationDirection: reverse`